### PR TITLE
therock.parseOrder string math

### PR DESCRIPTION
**Pull request by samgermain**
```
Node.js: v18.4.0
CCXT v2.5.17
therock.createOrder (BTC/USDT, market, sell, 0.00082772)
2023-01-04T02:28:30.968Z iteration 0 passed in 694 ms

{
  id: '1573319996',
  clientOrderId: undefined,
  info: {
    id: '1573319996',
    fund_id: 'BTCUSDT',
    side: 'sell',
    type: 'market',
    status: 'executed',
    price: '16665.38',
    amount: '0.0008',
    amount_unfilled: '0.0',
    conditional_type: null,
    conditional_price: null,
    date: '2023-01-04T02:28:30.651Z',
    close_on: null,
    dark: false,
    leverage: '1.0',
    position_id: null,
    client_uuid: null,
    trades: [
      {
        id: '10696135',
        fund_id: 'BTCUSDT',
        amount: '0.0008',
        price: '16665.38',
        side: 'sell',
        dark: false,
        date: '2023-01-04T02:28:30.887Z'
      }
    ]
  },
  timestamp: 1672799310651,
  datetime: '2023-01-04T02:28:30.651Z',
  lastTradeTimestamp: 1672799310887,
  status: 'closed',
  symbol: 'BTC/USDT',
  type: 'market',
  timeInForce: 'IOC',
  postOnly: undefined,
  side: 'sell',
  price: 16665.38,
  stopPrice: undefined,
  triggerPrice: undefined,
  cost: 13.332304,
  amount: 0.0008,
  filled: 0.0008,
  average: 16665.38,
  remaining: 0,
  fee: undefined,
  trades: [
    {
      info: {
        id: '10696135',
        fund_id: 'BTCUSDT',
        amount: '0.0008',
        price: '16665.38',
        side: 'sell',
        dark: false,
        date: '2023-01-04T02:28:30.887Z'
      },
      id: '10696135',
      order: undefined,
      timestamp: 1672799310887,
      datetime: '2023-01-04T02:28:30.887Z',
      symbol: 'BTC/USDT',
      type: undefined,
      side: 'sell',
      takerOrMaker: undefined,
      price: 16665.38,
      amount: 0.0008,
      cost: 13.332304,
      fee: { cost: undefined },
      fees: []
    }
  ],
  fees: []
}
2023-01-04T02:28:30.968Z iteration 1 passed in 694 ms
```

### CODE REVIEW

File: js/btcalpha.js [hunk 1]
```diff
@@ -645,7 +645,8 @@ module.exports = class btcalpha extends Exchange {
             throw new InvalidOrder (this.id + ' ' + this.json (response));
         }
         const order = this.parseOrder (response, market);
-        amount = (order['amount'] > 0) ? order['amount'] : amount;
+        const orderAmount = order['amount'].toString ();
```
Comment (carlosmiei) [inline js/btcalpha.js:648]: @samgermain is this a leftover or something?

If not, we can't do `order['amount'].toString ();` or even extend a parsed order with a stringified value

Comment (samgermain) [reply js/btcalpha.js:648]: > @samgermain is this a leftover or something?

What do you mean when you say that?

> If not, we can't do `order['amount'].toString ();` or even extend a parsed order with a stringified value

why not?

Comment (samgermain) [reply js/btcalpha.js:648]: @carlosmiei 👋

**Comment by frosty00** [comment]
@samgermain, this is the same issue as with #15003 , can you edit it to remove all the arithmetic?

**Pull request merged**